Source/Venice/Channel/Channel.swift

```diff
@@ -67,26 +67,23 @@ public final class Channel<T>: Sequence {
     }
 
     /// Closes the channel. When a channel is closed it cannot receive values anymore.
-    public func close() -> Bool {
-        if closed {
-            return false
-        }
+    public func close() throws {
+        guard !closed else { return }
```

Comment:
Yeah we should definitely have a more generic `Stream` protocol which both `Channel` and `BinaryStream` conform to. I agree that associated types complicate things but I'm thinking that in most cases inside C7 people would reference the actual `BinaryStream` (which doesn't have associated type requirements) instead of the generic `Stream`.